src/FluentDriver.php

```diff
@@ -117,6 +127,103 @@ public function getMappers()
     }
 
     /**
+     * Add mappings from an array of folders.
+     *
+     * @param string[] $paths
+     *
+     * @throws MappingException
+     */
+    public function addPaths($paths)
+    {
+        foreach ($paths as $path) {
+            if (!is_dir($path)) {
+                throw MappingException::fileMappingDriversRequireConfiguredDirectoryPath($path);
+            }
+
+            $iterator = new \RecursiveIteratorIterator(
+                new \RecursiveDirectoryIterator($path),
+                \RecursiveIteratorIterator::LEAVES_ONLY
+            );
+
+            foreach ($iterator as $file) {
+                if ($file->getBasename('.php') == $file->getBasename()) {
+                    continue;
+                }
+
+                $sourceFile = realpath($file->getPathName());
+                $className = $this->getClassFromFile($sourceFile);
+                $this->addMapping(new $className());
+            }
+        }
+    }
+
+    /**
+     * Get the FQN of a class from a source file.
+     *
+     * @param $pathToFile
+     *
+     * @return string
+     */
+    private function getClassFromFile($pathToFile)
+    {
+        //http://jarretbyrne.com/2015/06/197/
+        //Grab the contents of the file
+        $contents = file_get_contents($pathToFile);
+
+        //Start with a blank namespace and class
+        $namespace = $class = '';
+
+        //Set helper values to know that we have found the namespace/class token and need to collect the string values after them
+        $getting_namespace = $getting_class = false;
+
+        //Go through each token and evaluate it as necessary
+        foreach (token_get_all($contents) as $token) {
```

Comment:
The alternative is adding 80+ lines to my doctrine config, for each manager even. It gets unwieldy.  Plus with paths I Don't have to update my doctrine config each time I add a new mapping, just add the mapping and be done with it. 

I'm also not a huge fan of tokenizing but I couldn't get the doctrine way of doing it to work [using ReflectionClass() and requiring each file](http://www.doctrine-project.org/api/common/2.4/source-class-Doctrine.Common.Persistence.Mapping.Driver.StaticPHPDriver.html#82-133) which seems like way more overkill than tokenizing..